flocker/node/_loop.py

```diff
@@ -391,8 +393,16 @@ def output_UPDATE_MAYBE_WAKEUP(self, context):
         # still indicates some action should be taken that means we should
         # wake up:
         discovered = self._last_discovered_local_state
-        if self.deployer.calculate_changes(
-                self.configuration, self.cluster_state, discovered) != NoOp():
+        try:
+            changes = self.deployer.calculate_changes(
+                self.configuration, self.cluster_state, discovered)
+        except:
+            # Something went wrong in calculation due to a bug in the
+            # code. We should wake up just in case in order to be more
+            # responsive.
+            write_traceback()
+            changes = None
```

Comment:
Longer term I think we want any `eliot:traceback` message to make build go red for acceptance tests, so I don't think we need to do anything special h